### ISSUE TITLE
ttnn.max reduction operation with int32 datatype produces incorrect results

### ISSUE THREAD

**Issue by karthikkattaopenxpu** [tenstorrent/tt-metal#31295]
### Describe the bug

When performing a max reduction operation on an int32 tensor along a specific dimension using ttnn.max, the output does not match PyTorch's expected behavior.

### Steps to reproduce the issue

[reduce_max.txt](https://github.com/user-attachments/files/23180233/reduce_max.txt)

### Expected behavior

TTNN output : tensor([0], dtype=torch.int32)
Torch output : tensor([[418]], dtype=torch.int32)

**Comment by edwinleeTT** [comment]
Hi @karthikkattaopenxpu, I'm confused about the "Expected Behaviour" provided in this issue. It looks like you are expecting these tensors to be different, with TTNN producing `[0]` and Torch producing `[[418]]`? Not sure if this is intended, can you explain why these should differ? 

I looked at your repro code, and two things stand out. The first is that you are passing `keepdim=True` to the `torch.max()` call, but then omitting the argument for `ttnn.max()`. You also aren't specifying a dtype when converting the TTNN output back into Pytorch. Here's the slightly updated code, note the two changes with comments:

>     # Create input tensor with randn and convert to int32
>     torch_a = torch.randn((1, 128*1024), dtype=torch.float32)
>     torch_a = (torch_a * 100).to(torch.int32)
> 
>     # Compute expected output using PyTorch with dimension
>     torch_output, _ = torch.max(torch_a, dim=1, keepdim=True)
> 
>     # Convert to ttnn and run max with dimension
>     a = ttnn.from_torch(torch_a, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
>     output = ttnn.max(a, dim=1, keepdim=True) # must specify keepdim=True if we want to match Torch call
> 
>     # Convert back and compare
>     output_torch = ttnn.to_torch(output, dtype=torch.int32) #specify int32 here to make sure dtypes match
> 
>     print("TTNN Output:")
>     print(output_torch)
> 
>     print("Torch Output:")
>     print(torch_output)

This is my output (the tensor is random, but my random tensor happens to have 456 as the output of `max`):
> TTNN Output:
> tensor([[456]], dtype=torch.int32)
> Torch Output:
> tensor([[456]], dtype=torch.int32)

These outputs match, so this is performing as I would expect. Can you please try this out, and let me know if there are still any concerns?

**Comment by bbradelTT** [comment]
@edwinleeTT 
The question is why is the output value not the same. That is, the expected result is 
```
TTNN output : tensor([418], dtype=torch.int32)
Torch output : tensor([[418]], dtype=torch.int32)
```
and the actual result is 
```
TTNN output : tensor([0], dtype=torch.int32)
Torch output : tensor([[418]], dtype=torch.int32)
```

**Comment by bbradelTT** [comment]
@edwinleeTT 
Wouldn't
```
a = ttnn.from_torch(torch_a, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
```
be float32 instead of int32?

**Comment by karthikkattaopenxpu** [comment]
```
import torch
import ttnn    
  
device_id = 0
device = ttnn.open_device(device_id=device_id)

# Create input tensor with randint for int32  
torch_a = torch.randint(low=-100, high=100, size=(1, 128*1024), dtype=torch.int32)

# Compute expected output using PyTorch with dimension  
torch_output, _ = torch.max(torch_a, dim=1, keepdim=True)

# Convert to ttnn and run max with dimension  
a = ttnn.from_torch(torch_a, dtype=ttnn.int32, layout=ttnn.TILE_LAYOUT, device=device)
output = ttnn.max(a, dim=1, keepdim=True)

# Convert back and compare  
output_torch = ttnn.to_torch(output)

print(output_torch)
print(torch_output)
```

**TTNN output :** tensor([[0]], dtype=torch.int32)
**Torch output :** tensor([[99]], dtype=torch.int32)

@bbradelTT I think this test case eliminates the confusion and describes the problem better.. could you look into this ?

**Comment by edwinleeTT** [comment]
@karthikkattaopenxpu sorry for the confusion and thanks for the clarification. Currently the generic reduces don't support integer data types, and this is an issue with the op's input validation and docs. We will look into this further and scope out the possibility for integer support here, and if this is not possible, update the validation so that integer inputs error out cleanly.

**Comment by aczajkowskiTT** [comment]
@edwinleeTT Based on your comment above, are we moving this to a P2?

**Comment by edwinleeTT** [comment]
@aczajkowskiTT that sounds fair since this isn't high priority right now, and validation has been added to prevent this integer inputs
